### PR TITLE
Add Python 3.12-dev to tests actions

### LINKED ISSUE #326: Add Python 3.12-dev to tests actions
We should add Python 3.12-dev here and see if all tests pass: https://github.com/microsoft/vscode-isort/blob/main/.github/workflows/pr-check.yml

### PR DESCRIPTION
**Pull request by pheonix-18**
closes: https://github.com/microsoft/vscode-isort/issues/326, Added Python 3.12 Version to file:  https://github.com/microsoft/vscode-isort/blob/main/.github/workflows/pr-check.yml

### CODE REVIEW

**Comment by pheonix-18** [comment]
@karthiknadig I'm not sure how to run tests on my PR. 
Also can you review my PR title and commit message. 

Thank You

**Comment by anthonykim1** [comment]
Hello! @pheonix-18, I think it should be '3.12-dev' instead of '3.12'

**Comment by anthonykim1** [comment]
> @karthiknadig I'm not sure how to run tests on my PR. Also can you review my PR title and commit message.
I can help you run the test after we fix the 3.12 to 3.12-dev :)

File: .github/workflows/pr-check.yml [hunk 1]
```diff
@@ -45,7 +45,7 @@ jobs:
       fail-fast: false
       matrix:
         os: [ubuntu-latest, windows-latest]
-        python: ['3.7', '3.8', '3.9', '3.10', '3.11']
+        python: ['3.7', '3.8', '3.9', '3.10', '3.11', '3.12']
```
Comment (anthonykim1) [inline .github/workflows/pr-check.yml]: should be '3.12-dev' instead of '3.12'

**Review by anthonykim1 — CHANGES_REQUESTED** [review]

File: .github/workflows/pr-check.yml (hunk 1 above)
Comment (pheonix-18) [reply .github/workflows/pr-check.yml]: @anthonykim1 Fixed it, thank you for the suggestion.

**Review by anthonykim1 — APPROVED** [review]

**Review by anthonykim1 — APPROVED** [review]

**Review by eleanorjboyd — APPROVED** [review]

**Pull request merged**